test/new-e2e/tests/agent-metric-logs/linux-log-agent/file_tailing_utils_test.go

```diff
@@ -0,0 +1,178 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	componentos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/stretchr/testify/assert"
+)
+
+// generateLog generates x content with y reccurrence and verifies log contents.
+func generateLog(s *LinuxVMFakeintakeSuite, content string, recurrence int) {
+	// Determine the OS and set the appropriate log path and command.
+	var logPath, cmd, checkCmd string
+	t := s.T()
+	t.Helper()
+
+	osType := s.Env().VM.GetOSType()
+	var os string
+
+	switch osType {
+	case componentos.WindowsType:
+		os = "windows"
+		t.Log("Generating Windows log.")
+		logPath = "C:\\logs\\hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content+" ", recurrence), logPath)
```

Comment:
This will overwrite the contents of the existing file, not append. Not a blocking comment for this PR as this is focused on linux, but it should be fixed for the windows version.